src/status_im2/contexts/chat/lightbox/top_view.cljs

```diff
@@ -39,14 +42,33 @@
         (anim/animate top-view-width screen-width)
         (anim/animate top-view-bg colors/neutral-100-opa-0)))))
 
+(defn drawer
+  [uri]
+  [quo/action-drawer
+   [[{:icon                :i/save
+      :accessibility-label :save-image
+      :label               (i18n/label :t/save-image-library)
+      :on-press            (fn []
+                             (rf/dispatch [:hide-bottom-sheet])
+                             (images/save-image-to-gallery
+                              uri
+                              #(rf/dispatch [:toasts/upsert
+                                             {:id              :random-id
+                                              :icon            :correct
+                                              :icon-color      colors/success-50
+                                              :container-style {:bottom (when platform/android? 20)}
+                                              :text            (i18n/label :t/photo-saved)}])))}]]])
+
 (defn top-view
-  [{:keys [from timestamp]} insets index animations derived landscape? screen-width]
+  [{:keys [from timestamp content]} insets index animations derived landscape? screen-width]
   (let [display-name                       (first (rf/sub [:contacts/contact-two-names-by-identity
                                                            from]))
         bg-color                           (if landscape?
                                              colors/neutral-100-opa-70
                                              colors/neutral-100-opa-0)
-        {:keys [background-color opacity]} animations]
+        {:keys [background-color opacity]} animations
+        uri                                (http/replace-port (:image content)
```

Comment:
it's better to move this into drawer, because it won't be updated if app will be hidden with bottom sheet opened